cartography/intel/entra/applications.py

```diff
@@ -235,10 +245,48 @@ def transform_app_role_assignments(
     return result
 
 
+def transform_service_principals(
+    service_principals: list[ServicePrincipal],
```

Comment:
@shyammukund this is what I was talking about at lunch: use `list[SpecificType]` wherever possible (like here with the Entra SDK because it's available to us. This way we can do things like `spn.login_url` instead of `getattr(spn, 'login_url', None)`